src/main/java/seedu/address/logic/commands/RequiredCommand.java

```diff
@@ -0,0 +1,194 @@
+     */
+    public boolean doesModuleAlreadyExist(Module module, ObservableList<Module> modules) {
+        for (Module mod : modules) {
+            if (module.isSameModule(mod)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Cross references the user's current list of Modules against the given
+     * modules argument and marks out any undone Modules. Displays a successMessage
+     * if all modules are done, and a failMessage if there are left over modules.
+     * @param modules List of Modules of a certain category (Eg. Foundation, IT Professsionalism)
+     * @param failMessage Fail message for particular category of Modules.
+     * @param successMessage Success message for particular category of Modules.
+     */
+    public void compareModules(ObservableList<Module> modules, String failMessage, String successMessage) {
+        boolean areModulesCleared = true;
+        StringBuilder modulesToAdd = new StringBuilder();
+        for (Module module : modules) {
+            if (!doesModuleAlreadyExist(module, currentModules)) {
+                String moduleToAdd = module.getModuleCode() + " (" + module.getModularCredits() + " MCs)";
+                modulesToAdd.append("\n").append(moduleToAdd);
+                areModulesCleared = false;
+            }
+        } if (areModulesCleared) {
+            leftOverModules += successMessage + "\n";
+        } else {
+            leftOverModules += failMessage + modulesToAdd + "\n";
+        }
+        leftOverModules += "\n";
+    }
+
+    /**
+     * Cross references the user's current list of Modules and marks out
+     * any undone Science Modules.
+     */
+    public void compareScience(ObservableList<Module> requiredScience) {
+        boolean areModulesCleared = true;
+        for (Module module : requiredScience) {
+            if (doesModuleAlreadyExist(module, currentModules)) {
+                leftOverModules += MESSAGE_SUCCESS_SCIENCE + "\n";
+                areModulesCleared = false;
+                break;
+            }
+        } if (areModulesCleared) {
+            leftOverModules += MESSAGE_SCIENCE + "\n";
+        }
+        leftOverModules += "\n";
+    }
+
+    /**
+     * Cross references the user's current list of Modules and marks out
+     * any undone Internship Modules. Also calculates current MC score
+     * achieved from Internship Modules.
+     */
+    public void compareInternship(ObservableList<Module> requiredInternship) {
+        int modularScore = 0;
+        StringBuilder leftOverInternship = new StringBuilder();
+        for (Module module : requiredInternship) {
+            boolean areModulesCleared = true;
+            if (doesModuleAlreadyExist(module, currentModules)) {
+                int modularCredits = Integer.parseInt(module.getModularCredits().toString());
+                modularScore += modularCredits;
+                areModulesCleared = false;
```

Comment:
How come taking a single internship module means that `areModulesCleared` is false ahh? I tot it should only be false if total MCs < 12 since we need >= 12 in order to clear the internship basket ><

 I FEEL LIKE IM MISSING SOMETHING OBVIOUS LOL SO PLS FORGIVE ME IF I AM HAHA IM DAMN CONFUSED

Reply:
sorry for the misleading boolean names AHAHHA changed the whole code, so no more booleans, just an If-else block to settle if the module is present or not